src/paulscode/android/mupen64plusae/input/map/PlayerMap.java

```diff
@@ -79,7 +88,9 @@ public String getDeviceSummary( Context context, int player )
             if( mMap.valueAt( i ) == player )
             {
                 int deviceId = mMap.keyAt( i );
-                String name = AbstractProvider.getHardwareName( deviceId );
+                String name = AbstractProvider.isHardwareAvailable( deviceId ) ?
+                        AbstractProvider.getHardwareName( deviceId ) :
+                        "Not Connected";
```

Comment:
Here is how you can reproduce it. Install the latest auto build and map a couple of controllers to players. Then install your version of the build and do the same again. You will now have 2 controllers for each player with 1 "not connected".